src/conn.cc

```diff
@@ -1765,6 +1777,10 @@ static void connect_failed_try_next(drizzle_st *con, const char *file, uint line
   drizzle_set_error(con, file, line, function, "connect: %s (port %s): %s",
                     hostbuf, servbuf, msg);
 
-  con->addrinfo_next= con->addrinfo_next->ai_next;
+  if (con->addrinfo_next != nullptr)
```

Comment:
Shouldn't you also be returning `DRIZZLE_RETURN_COULD_NOT_CONNECT` if this fails?

Reply:
I don't reckon so. But I did a modification of the PR where this is added : 
```c++
static void connect_failed_try_next(drizzle_st *con, const char *file, uint line,
  const char *function, const char *msg)
{
  ...

  if (con->addrinfo_next == nullptr)
  {
    drizzle_log_info(con, __FILE_LINE_FUNC__, "refreshing addresses");
    con->push_state(drizzle_state_connect);
    con->push_state(drizzle_state_addrinfo);
  }
  else
  {
    con->addrinfo_next= con->addrinfo_next->ai_next;
    con->push_state(drizzle_state_connect);
  }
}
```

if the addrinfo is NULL `drizzle_state_addrinfo` would try to refresh it before `drizzle_state_connect` is entered again. If `drizzle_state_connect` connects we would with the changes return **DRIZZLE_RETURN_COULD_NOT_CONNECT** to the client

Reply:
Ah, right. This should have been left as it was initially.